packages/client/hmi-client/src/workflow/tera-workflow.vue

```diff
@@ -880,7 +880,9 @@ function createNewEdge(node: WorkflowNode<any>, port: WorkflowPort, direction: W
 }
 
 function removeEdge(portId: string) {
-	const edge = wf.value.edges.find(({ targetPortId }) => targetPortId === portId);
+	const edge = wf.value.edges.find(
+		({ targetPortId, sourcePortId }) => targetPortId === portId || sourcePortId === portId
+	);
 	if (edge) workflowService.removeEdge(wf.value, edge.id);
 	else logger.error(`Edge with port id:${portId} not found.`);
 }
```

Comment:
This function appears to have a split-brain, your logics would seeming indicate you want to remove multiple edges, but your execution logic only removes a single edge. What exactly do you want to do?

Imagine a Y-junction and you click "unlink" at "A"
```
         /----  X
A ==  
        \ ----- Y
```

Reply:
When this is called from `operator-inputs` we want to match the `portId` with `targetPortId`. With `operator-outputs` we looking to match `sourcePortId`. I'll split those up more.

Reply:
I meant you seemingly want to match multiple edges, but your code

```
if (edge) workflowService.removeEdge(wf.value, edge.id);
```

removes only one, I am not sure if you want to remove a single-edge? or all-edges attached to a particular port-id.

Reply:
I thought for inputs I'd remove one and for outputs I'd remove all. I assumed `removeEdge` in `services/workflow.ts` removes all edges specifically for outputs but I read that wrong. Do you think it's fine to keep it removing one edge for now or should I edit `removeEdge` to remove all edges related to the one port id?

Reply:
I see, yeah we can edit removeEdge in "tera-workflow", I guess you will want something like this:

```
const edges = wf.value.edges.filter(...)
edges.forEach(edge => {
  workflowService.removeEdge(wf.value, edge.id);  
});

```